File: AngularFrontend/src/app/types/employee.ts
export interface IEmployee {
 id:number;
  name: string;
  email: string;
  phone: string;
  jobTitle: string;
  gender: Gender;
  departmentId: number | null;   // nullable
  userId?: number | null;      
  joiningDate: string;
 department:string;
  lastWorkingDate: string;
  dateOfBirth: string;
  
 
}

export enum Gender {
  Male = 0,
  Female = 1,
}

File: Asp.NetCoreWebAPI/EmployeeManagementSystem/Entity/Employee.cs
﻿using System.ComponentModel.DataAnnotations.Schema;

namespace EmployeeManagementSystem.Entity
{
    public class Employee
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public string Phone {  get; set; }
        public string JobTitle {  get; set; }
        public int Gender {  get; set; }
        [ForeignKey(nameof(Department))]
        public int? DepartmentId {  get; set; }
        public Department? Department { get; set; }
        public DateTime JoiningDate {  get; set; }
        public DateTime LastWorkingDate { get; set; }
        public DateTime DateOfBirth { get; set; }   

    }
}
